model/textparse/protobufparse.go

```diff
@@ -299,21 +304,8 @@ func (p *ProtobufParser) Comment() []byte {
 // Metric writes the labels of the current sample into the passed labels.
 // It returns the string from which the metric was parsed.
 func (p *ProtobufParser) Metric(l *labels.Labels) string {
-	p.builder.Reset()
-	p.builder.Add(labels.MetricName, p.getMagicName())
-
-	for _, lp := range p.mf.GetMetric()[p.metricPos].GetLabel() {
-		p.builder.Add(lp.GetName(), lp.GetValue())
-	}
-	if needed, name, value := p.getMagicLabel(); needed {
-		p.builder.Add(name, value)
-	}
-
-	// Sort labels to maintain the sorted labels invariant.
-	p.builder.Sort()
-	*l = p.builder.Labels()
-
-	return p.metricBytes.String()
+	*l = p.lset.Copy()
+	return yoloString(p.seriesBytes.Bytes())
```

Comment:
This is unsafe.  Since nobody is using it, perhaps just return "" ?
Alternatively the docstring should say not to retain it, or go further and remove the return value (as you commented).

Reply:
Fair, changing `Metric` method in https://github.com/prometheus/prometheus/pull/16012